src/App/App.tsx

```diff
@@ -462,8 +461,7 @@ const App = (props: AppProps) => {
                   // tslint:disable-next-line: jsx-no-lambda
                   component={() => {
                     if (BACKEND_ACTIVE) {
-                      store.dispatch(logOutUser());
```

Comment:
Hmm, does removing this completely make things work better?

Reply:
I added some information at the time of making this change; on why I thought it should be made
> I believe the dispatch action causes any views dependent on the gatekeeper authenticated state to rerender, which is I think the whole app. this introduces an unforseen and uncotrolled rerender that was not initially accounted for when writing the logout component for authorization code flow. 

this rerender actually makes things a little harder to think about, but it seems when I removed the dispatch action, the logout component was able to execute to completion. 
The things that I am not quite certain of is how the gatekeeper module `authenticated` variable resets to false,.. My initial thought was that after logout; when `fetchState` is invoked while attempting to access any of the connected private routes, it finds no session info and rehydrates the store to reflect that.

I also think if that if we are going to treat this in the SSO line of thought, the opensrp server should be able to redirect the user to a predefined url, in this case the express logout url.

Reply:
@moshthepitt :point_up: